src/Microsoft.Identity.Web.UI/ServiceCollectionExtensions.cs

```diff
@@ -17,6 +18,11 @@ public static class ServiceCollectionExtensions
         /// <param name="builder">MVC Builder.</param>
         public static IMvcBuilder AddMicrosoftIdentityUI(this IMvcBuilder builder)
         {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
```

Comment:
Option 2.
Set the path inside `AddMicrosoftIdentityUI`. We need to set the path on the scheme with the same name that we assigned to `OpenIdConnectOptions.SignInScheme` in `AddSignIn`. That's why I added a scheme parameter. 
**Pro** UI related code is together.
**Con** It looks a bit weird passing a cookie scheme to this UI related method. 
**Con** If a user uses a non-default cookie scheme name, they would have to pass it here. A bit fragile.
```csharp
public static IMvcBuilder AddMicrosoftIdentityUI(this IMvcBuilder builder, string cookieScheme = CookieAuthenticationDefaults.AuthenticationScheme)
{
	if (builder == null)
	{
		throw new ArgumentNullException(nameof(builder));
	}

	builder.ConfigureApplicationPartManager(apm =>
	{
		apm.FeatureProviders.Add(new MicrosoftIdentityAccountControllerFeatureProvider());
	});

	builder.Services.Configure<CookieAuthenticationOptions>(cookieScheme, options =>
	{
		options.AccessDeniedPath = new PathString("/MicrosoftIdentity/Account/AccessDenied");
	});

	return builder;
}
```

Reply:
That's my prefered option (even if I understand the fragility), because /MicrosoftIdentity/Account/AccessDenied is defined in Microsoft.Identity.Web.UI, which customers are not obliged to use.

We'll document the fragility


Is there a way for us to understand which CookieScheme would have been passed into AddSignIn ?

Reply:
I think we could check the `Services` here if the cookie or OIDC options were added. But that won't work if `AddSignIn` is called after this method.

I also tried configuring without the scheme name, which should be the default, but that didn't work. 

I did have an idea to maybe use PostConfigure somehow, which is trigged after all of the options are added. Will have to explore.

Reply:
@jmprieur `ConfigureAll` works. No need to specify scheme name for this.